expr/functions/removeEmptySeries/function.go

```diff
@@ -55,8 +60,11 @@ func (f *removeEmptySeries) Do(ctx context.Context, e parser.Expr, from, until i
 				}
 			}
 		}
-		if nonNull != 0 && nonNull/float64(len(arg.Values)) >= factor {
-			results = append(results, arg)
+
+		if nonNull != 0 && helper.XFilesFactor(nonNull, len(arg.Values), xFilesFactor) {
```

Comment:
May be add some tests for XFilesFactor ?

Reply:
It looks like there are a few tests using XFilesFactor for removeEmptySeries and one for removeZeroSeries. Would you like us to add a few more tests?

Reply:
May be not now. It's not blocking.